.github/workflows/ci-lint-go.yml

```diff
@@ -0,0 +1,63 @@
+name: Run lint for a Go project
+run-name: "${{ inputs.project-directory }}"
+
+on:
+  workflow_call:
+    inputs:
+      project-directory:
+        required: true
+        type: string
+        default: "."
+        description: "The directory where the Go project is located."
+
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+
+jobs:
+  lint-go-project:
+    name: "lint: ${{ inputs.project-directory }}"
+    runs-on: 'ubuntu-latest'
+    continue-on-error: false
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Set up Go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5
+        with:
+            go-version-file: "${{ inputs.project-directory == '' && '.' || inputs.project-directory }}/go.mod"
+            cache-dependency-path: "${{ inputs.project-directory == '' && '.' || inputs.project-directory }}/go.sum"
+        id: go
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@971e284b6050e8a5849b72094c50ab08da042db8 # v6.1.1
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          version: v1.61.0
+          # Optional: working directory, useful for monorepos
+          working-directory: ${{ inputs.project-directory }}
+          # Optional: golangci-lint command line arguments.
+          args: --verbose
+          # Optional: if set to true then the all caching functionality will be complete disabled,
+          #           takes precedence over all other caching options.
+          skip-cache: true
+
+      - name: generate
+        working-directory: ./${{ inputs.project-directory }}
+        shell: bash
+        run: |
+          make generate
+          git --no-pager diff && [[ 0 -eq $(git status --porcelain | wc -l) ]]
+
+      - name: modVerify
+        working-directory: ./${{ inputs.project-directory }}
+        run: go mod verify
```

Comment:
question: is this really needed as CI is clean each time it starts?

Reply:
I'm removing it elsewhere but in the windows workers for the Docker infra